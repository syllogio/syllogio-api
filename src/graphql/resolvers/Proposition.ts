import { map } from 'ramda';
import { getProposition, IProposition } from '../../lib/proposition';

function supports(proposition: IProposition) {
  return map(getProposition, proposition.supports);
}

function supportedBy(proposition: IProposition) {
  return map(getProposition, proposition.supportedBy);
}

export default {
  supportedBy,
  supports,
};
