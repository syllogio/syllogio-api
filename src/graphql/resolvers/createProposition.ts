import {
  createProposition as _createProposition,
  INewProposition,
  IProposition,
} from '../../lib/proposition';

const DEFAULT_PROPS = {
  supportedBy: [],
  supports: [],
};

/**
 * GraphQL resolver for the createProposition mutation.
 */
export default async function createProposition(
  _: any,
  args: { text: string; supports?: number[]; supportedBy?: number[] }
): Promise<IProposition> {
  return _createProposition({ ...DEFAULT_PROPS, ...args });
}
