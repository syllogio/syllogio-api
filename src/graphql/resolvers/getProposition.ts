import {
  getProposition as _getProposition,
  IProposition,
} from '../../lib/proposition';

/**
 * GraphQL resolver for the proposition field.
 */
export default async function getProposition(
  _: any,
  { id }: { id: number }
): Promise<IProposition | void> {
  return _getProposition(id);
}
