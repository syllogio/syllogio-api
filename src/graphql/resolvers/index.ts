import proposition from './proposition';

export default {
  Query: {
    proposition,
  },
};
