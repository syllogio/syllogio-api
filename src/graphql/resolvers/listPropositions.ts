import { IProposition, listPropositions } from '../../lib/proposition';

/**
 * GraphQL resolver for the propositions field.
 */
export default async function propositions(): Promise<IProposition[]> {
  return listPropositions();
}
