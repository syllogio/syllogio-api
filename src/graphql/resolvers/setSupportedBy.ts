import {
  IProposition,
  setSupportedBy as _setSupportedBy,
} from '../../lib/proposition';

/**
 * GraphQL resolver that allows for supportedBys to be mutated.
 */
export default async function setSupportedBy(
  _: any,
  {
    propositionId,
    supportedBy,
  }: {
    propositionId: number;
    supportedBy: number[];
  }
): Promise<boolean> {
  await _setSupportedBy(propositionId, supportedBy);
  return true;
}
