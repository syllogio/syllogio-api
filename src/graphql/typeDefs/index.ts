import { importSchema } from 'graphql-import';
import { join } from 'path';

const typeDefs = importSchema(join(__dirname, './Query.graphql'));

export default typeDefs;
