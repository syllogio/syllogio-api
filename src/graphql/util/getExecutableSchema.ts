import { GraphQLSchema } from 'graphql';
import { importSchema } from 'graphql-import';
import { makeExecutableSchema } from 'graphql-tools';

export default async function getExecutableSchema(): Promise<GraphQLSchema> {
  return makeExecutableSchema({
    typeDefs: importSchema('../../graphql/typeDefs/Query.graphql'),
  });
}
