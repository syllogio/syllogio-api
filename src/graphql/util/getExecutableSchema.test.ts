import { makeExecutableSchema } from 'graphql-tools';
import resolvers from '../resolvers';
import typeDefs from '../typeDefs';
import getExecutableSchema from './getExecutableSchema';

jest.mock('graphql-tools');

(makeExecutableSchema as jest.Mock<any>).mockResolvedValue({ the: 'schema' });

describe('getExecutableSchema', () => {
  it('retuns an executable schema', async () => {
    expect.assertions(2);
    await expect(getExecutableSchema()).resolves.toMatchSnapshot();
    expect(makeExecutableSchema).toHaveBeenCalledWith({
      resolvers,
      typeDefs,
    });
  });
});
