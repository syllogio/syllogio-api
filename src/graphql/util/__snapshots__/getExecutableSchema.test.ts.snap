// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getExecutableSchema retuns an executable schema 1`] = `
Object {
  "the": "schema",
}
`;
